src/coprocessor.rs

```diff
@@ -0,0 +1,75 @@
+use std::fmt::Debug;
+
+use crate::eval::IO;
+use crate::field::LurkField;
+
+use crate::store::{ContPtr, Ptr, Store};
+
+pub trait Coprocessor<F: LurkField>: Debug + Sync {
+    fn arity(&self) -> usize;
+
+    fn evaluate(&self, s: &mut Store<F>, args: Ptr<F>, env: Ptr<F>, cont: ContPtr<F>) -> IO<F> {
+        let argv: Vec<Ptr<F>> = if let Some(list) = s.fetch_list(&args) {
+            list
+        } else {
+            return IO {
+                expr: args,
+                env,
+                cont: s.intern_cont_error(),
+            };
+        };
+
+        assert_eq!(argv.len(), self.arity());
+
+        let result = self.simple_evaluate(s, &argv);
+
+        IO {
+            expr: result,
+            env,
+            cont,
+        }
+    }
+
+    fn simple_evaluate(&self, s: &mut Store<F>, args: &[Ptr<F>]) -> Ptr<F>;
+}
+
+#[cfg(test)]
+pub(crate) mod test {
+    use super::*;
+    use std::marker::PhantomData;
+
+    /// A dumb Coprocessor for testing.
```

Comment:
I’m not sure the adjective helps here. ` Trivial`, ` Testing` or ` Dummy` might be more useful.

Reply:
FWIW, Dummy shows up in the coming work in a more meaningful way. I can change to `Test` or `Stupid`.

Reply:
`Stupid` also comes with a qualitative connotation I’d like to avoid, but the functional `Test` seems fine.

Reply:
One problem with `test` is that it makes `test_test_lang`, etc. awkward. I'll change it if you feel strongly enough, but since it says exactly what I intend linguistically, and all other candidates that aren't synonyms (including `Stupid`) are less pareto-optimal, I'm inclined to leave it.

At the risk of a philosophical discussion, a name with qualitative connotations isn't itself problematic — from my perspective — when the intent is specifically to communicate those qualities. It serves no particular purpose, but it does have some properties we care about. It's entirely coherent to talk about 'dumbing things down', etc. — and we shouldn't particularly shy away from apposite use of language.

While this can be solved for, it's an extra constraint in naming space that doesn't add much in my opinion, which is why I decided to voice that.

Reply:
All that said, this is probably just me being cranky about needing to draw a box around work I wish could be more polished before merging — since there is pressure to get what is essential in quickly. So please feel more than free to disregard with prejudice. We can fix/change later.

Reply:
FWIW, I don’t think it’s a big deal.